src/cocotb/decorators.py

```diff
@@ -57,9 +57,16 @@ def function(func: Callable[..., Coroutine[Any, Any, Result]]) -> Callable[...,
     """Decorator that turns a :term:`coroutine function` into a blocking function.
 
     This allows a coroutine that consumes simulation time
-    to be called by a thread started with :class:`cocotb.external`;
-    in other words, to internally block while externally
-    appear to yield.
+    to be called by a thread started with :class:`cocotb.external`.
+    When the blocking function is called, the external thread will block until
+    the :class:`~cocotb.task.Task` containing the coroutine finishes.
+    The result from the coroutine will be returned to the caller of the blocking function.
+
+    Args:
+        func: The :term:`coroutine function` to wrap/convert.
+
+    Returns:
+        The function to be called.
```

Comment:
I think we want to add what is thrown if `cocotb.function` is called from a non-`external` thread.